### ISSUE TITLE
Rule blocked: CORERULES-9788 (TIG0535 and CG0371) - add 'within' functionality to operators like 'is_not_valid_relationship'

### ISSUE THREAD

**Issue by eljanssens** [cdisc-org/cdisc-rules-engine#1135]
Test data: 

[unit-test-coreid-CG0371-negative.xlsx](https://github.com/user-attachments/files/19759890/unit-test-coreid-CG0371-negative.xlsx)

**Links to related JIRA Tickets**

- https://jira.cdisc.org/browse/CORERULES-213
- https://jira.cdisc.org/browse/CORERULES-9706
- https://jira.cdisc.org/browse/CORERULES-570
- https://jira.cdisc.org/browse/CORERULES-9788

**Rule Information**
SEND, SDTM, TIG
- TIG0535
- CG0371
- SEND121
- TIG058
- 
# Variable: IDVARVAL
# Condition: IDVAR ^= null
# Rule: IDVARVAL = value of variable = IDVAR in domain = RDOMAIN

**Describe the problem**
Operator: is_valid_relationship and is_not_valid_relationship should also work 'within' a specified variable.

**Screenshots**

![Image](https://github.com/user-attachments/assets/1b09059f-b7b8-4671-a8e7-d1054496a86e)

**Comment by SFJohnson24** [comment]
@eljanssens this is merged now

**Comment by eljanssens** [comment]
@SFJohnson24 I checked this and it now gives correct output for negative test data but for the positive1 test data, I now get negative output which is not what I expect. Can you please have a look?

**Comment by SFJohnson24** [comment]
This ticket has moved back to in-progress.  The goal would be to remove is_valid_relationship and is_valid_reference operators and remove the relationship_data that is grabbed in the preprocessor.  We would offer better merge tooling allowing for this cross-domain check.

**Comment by eljanssens** [comment]
Since this ticket is now closed and indeed is_not_valid_relationship is not longer a valid operator (yellow sguiggly lines), is there any guidance on how to link the value of idvar and idvarval with the variable/value of the related domain (rdomain)? Should this just be possible via match datasets?

**Comment by SFJohnson24** [comment]
today we discussed one final ticket for this rule: the ability to do 'Supp--' for the match datasets logic.  This would allow for easier rule logic.  I am reopening this issue and the A/C is being able to run a rule with this logic for match dataset:
```yaml
    {
      "Name": "SUPP--",
      "Keys": ["USUBJID"],
      "Child": "True",
      "Is_Relationship": true
    },
```
